token/program-2022/src/processor.rs

```diff
@@ -298,24 +308,32 @@ impl Processor {
                 return Err(TokenError::MintDecimalsMismatch.into());
             }
 
-            if let Ok(transfer_fee_config) = mint.get_extension::<TransferFeeConfig>() {
+            let fee = if let Ok(transfer_fee_config) = mint.get_extension::<TransferFeeConfig>() {
                 transfer_fee_config
                     .calculate_epoch_fee(Clock::get()?.epoch, amount)
                     .ok_or(TokenError::Overflow)?
             } else {
                 0
-            }
+            };
+
+            let maybe_permanent_delegate = mint
+                .get_extension::<PermanentDelegate>()
+                .map(|e| Option::<Pubkey>::from(e.delegate))
+                .ok()
+                .flatten();
```

Comment:
im not requesting this change but ptsd from haskell compells me to note that the above (fmap and join) is equivalent to
```
            let maybe_permanent_delegate = mint
                .get_extension::<PermanentDelegate>()
                .ok()   
                .and_then(|e| Option::<Pubkey>::from(e.delegate)));
```

this (bind)

Reply:
That's probably better than using a `flatten()`, I always feel a bit icky when I do it